sys/net/gnrc/netif/ieee802154/gnrc_netif_ieee802154.c

```diff
@@ -90,7 +90,7 @@ static inline bool _already_received(gnrc_netif_t *netif,
 static gnrc_pktsnip_t *_recv(gnrc_netif_t *netif)
 {
     netdev_t *dev = netif->dev;
-    netdev_ieee802154_rx_info_t rx_info;
+    netdev_ieee802154_rx_info_t rx_info = { .crc_valid = 0 };
```

Comment:
Why is `crc_valid` set to 0? AFAIS this will be always set by the driver in this case.

Reply:
It is not guaranteed that the driver will set it. In that case not setting it to 0 would mean the value would be in an undefined state (as it is allocated on a dirty stack).

Reply:
I think the driver should always set it. If the driver doesn't set it, it will be seen as invalid by the MAC layer.

If the radio doesn't support CRC calculation, this should be ignored via radio_caps (see #11473 ). If the driver doesn't provide a flag for the CRC (but does the calculations), it can be set to valid anyway.

Reply:
In *this* PR we can't assume that and we also should not. Otherwise this PR needs to touch *all* the drivers.

Reply:
It is also in general not save to assume the callee changes a value, so it should *always* be set if it is checked later. Just from a security perspective ;-).

Reply:
(a proper static checker will complain about such things)

Reply:
> It is also in general not save to assume the callee changes a value, so it should always be set if it is checked later. Just from a security perspective ;-).

In that case I think `memset` is cleaner, considering there might be more rx_info stuff. This is independent to the fact the driver should always set it if it support automatic CRC calculation

Reply:
`memset()` does exactly the same thing as the line above. Just without type check ;-P.

> This is independent to the fact the driver should always set it if it support automatic CRC calculation

So you agree now, that we *can't* guarantee the driver will set it? ;-)

Reply:
Ok, after some offline discussion: would some documentation amendment e.g.

```C
    /**
     * @brief CRC was valid (when != 0)
     * @warning this is not implemented for all drivers yet,
     *          so some might not set it.
     */
    uint8_t crc_valid;
```

Reply:
> memset() does exactly the same thing as the line above. Just without type check ;-P.

I always meant `memset` for the whole structure, not only the `crc_valid` field. Just in case :)

> So you agree now, that we can't guarantee the driver will set it? ;-)

Yes, but the statement if incomplete to me. It should also include that "it's guaranteed that the driver will set the `crc_valid` if the radio supports CRC". Otherwise there will be false positives.

The upper layer has to know if the radio calculates the CRC or not anyway (via radio caps), so with these 2 statements it's possible to reliably indicate the upper layers about crc validity. 

Since the radio caps are still WIP, the documentation amendment proposed by @miri64 could do the trick.

Reply:
> I always meant memset for the whole structure, not only the crc_valid field. Just in case :)

I know :-)

Reply:
> memset() does exactly the same thing as the line above. Just without type check ;-P

We should set the whole `netdev_ieee802154_rx_info_t` struct to zero and not just `.crc` (which can be easily done using `memset` )

Reply:
Again... The initialization as used above does this implicitly, but yeah maybe using `memset()` is more clearer. At least that's how the discussion seems to go.

Reply:
Most of the  devices calculate the CRC themselves and don't (necessarily) expose it. Furthermore, oftentimes only frames with valid CRCs are even indicated by our drivers. In that case, `.crc` needs to be set to a value !=0. Where does this happen? Or did I misunderstand something?

Reply:
> In that case, .crc needs to be set to a value !=0. Where does this happen? Or did I misunderstand something?

It the drivers. However, this PR only ports the `cc2420`. That's why I and @jia200x suggested that @roberthartung adds the big fat warning to the doc of that field. Requiring all the drivers to be ported in this PR would explode this PR.

Reply:
> The initialization as used above does this implicitly

So you are saying that `netdev_ieee802154_rx_info_t rx_info = { .crc_valid = 0 };` initializes the whole structure with `0`? (And I'm not talking about some smart compiler things going on in the background)

Reply:
>  Requiring all the drivers to be ported in this PR would explode this PR.

Reply:
> So you are saying that `netdev_ieee802154_rx_info_t rx_info = { .crc_valid = 0 };` initializes the whole structure with `0`? (And I'm not talking about some smart compiler things going on in the background)

To say it semantically correct:  `struct foobar foo = { .field = x }` initializes `field` with `x` and all fields of `foobar` that are not provided with 0. So in the case above, yes it initializes the whole struct with 0 (AFAIK in the background most compilers solve this by just implicitly calling `memset()` though). See https://en.cppreference.com/w/c/language/struct_initialization for reference.

> How about we wait until someone opens the follow-up PR and then we both consecutively?

See #9755 for `at86rf2xx`